src/Microsoft.AspNet.Server.Kestrel/ServerFactory.cs

```diff
@@ -53,9 +54,16 @@ public IDisposable Start(IFeatureCollection serverFeatures, Func<IFeatureCollect
             try
             {
                 var information = (KestrelServerInformation)serverFeatures.Get<IKestrelServerInformation>();
-                var engine = new KestrelEngine(_libraryManager, new ServiceContext { AppShutdown = _appShutdownService, Log = new KestrelTrace(_logger) });
+                var dateHeaderValueManager = new DateHeaderValueManager();
+                var engine = new KestrelEngine(_libraryManager, new ServiceContext
+                {
+                    AppShutdown = _appShutdownService,
+                    Log = new KestrelTrace(_logger),
+                    DateHeaderValueManager = dateHeaderValueManager
+                });
 
                 disposables.Push(engine);
+                disposables.Push(dateHeaderValueManager);
```

Comment:
Yeah it's messy, but this ensures that "he who creates it disposes it". We should look at this all up later.